### ISSUE TITLE
bnd taking property defaults from component property type (the pre-spec ones) instead of properties on @Component

### ISSUE THREAD

**Issue by rotty3000** [bndtools/bnd#2110]
Given the following component:

```
@interface Conf {
    String fight_sound() default "BAM!";
}

@Component(property = "fight.sound=KAPOW!")
class ComicBattle {
    void activate(Conf conf) {
        assert conf.fight_sound() == "KAPOW!";
    }
}
```

The component currently fails! The reason is that bnd writes into the DS XML the value from `Conf` and not the value from `property`.

Is this a bug in the precedence ordering? Or is this expected? I wasn't sure... it seems wrong. What if the annotation is defined elsewhere, used by many components, and I want to override the default!

**Comment by bjhargrave** [comment]
The precedence rules are in the DS spec.

-- 
BJ

**Comment by rotty3000** [comment]
According to 112.8.3 it seems according to my observations that bnd is not applying the proper order.

**Comment by kwin** [comment]
Indeed the order being added with https://github.com/bndtools/bnd/issues/2163 seems wrong.

**Comment by timothyjward** [comment]
@rotty3000 you are correct that this is broken - it shouldn't take much effort to fix, and it looks like you have most of a failing test case. Any chance of a PR in time for 4.0?

**Comment by kwin** [comment]
It would be good to include in the same test a check for accessing the properties via `ComponentContext`.

**Comment by rotty3000** [comment]
Sadly, I'm unlikely to have time to make a fix before 4.0.0.
